components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/basicauth/BasicAuthCredentialValidator.java

```diff
@@ -0,0 +1,478 @@
+     *
+     * @return the invalid username cache
+     */
+    private Cache getInvalidUsernameCache() {
+        String apimGWCacheExpiry = getApiManagerConfiguration().
+                getFirstProperty(APIConstants.TOKEN_CACHE_EXPIRY);
+
+        if (!gatewayUsernameCacheInit) {
+            gatewayUsernameCacheInit = true;
+            if (apimGWCacheExpiry != null) {
+                return createCache(APIConstants.GATEWAY_INVALID_USERNAME_CACHE_NAME,
+                        Long.parseLong(apimGWCacheExpiry), Long.parseLong(apimGWCacheExpiry));
+            } else {
+                long defaultCacheTimeout = getDefaultCacheTimeout();
+                return createCache(APIConstants.GATEWAY_INVALID_USERNAME_CACHE_NAME,
+                        defaultCacheTimeout, defaultCacheTimeout);
+            }
+        }
+        return getCacheFromCacheManager(APIConstants.GATEWAY_INVALID_USERNAME_CACHE_NAME);
+    }
+
+    /**
+     * Create the Cache object from the given parameters.
+     *
+     * @param cacheName   name of the Cache
+     * @param modifiedExp value of the modified expiry type
+     * @param accessExp   value of the accessed expiry type
+     * @return the cache object
+     */
+    private Cache createCache(final String cacheName, final long modifiedExp,
+                              long accessExp) {
+        return APIUtil.getCache(APIConstants.API_MANAGER_CACHE_MANAGER, cacheName, modifiedExp, accessExp);
+    }
+
+    /**
+     * Returns the API Manager Configuration.
+     *
+     * @return the API Manager Configuration
+     */
+    private APIManagerConfiguration getApiManagerConfiguration() {
+        return ServiceReferenceHolder.getInstance().getAPIManagerConfiguration();
+    }
+
+    /**
+     * Returns the Cache object of the given name.
+     *
+     * @param cacheName name of the Cache
+     * @return the cache object
+     */
+    private Cache getCacheFromCacheManager(String cacheName) {
+        return Caching.getCacheManager(
+                APIConstants.API_MANAGER_CACHE_MANAGER).getCache(cacheName);
+    }
+
+    /**
+     * Returns the default cache timeout.
+     *
+     * @return the default cache timeout
+     */
+    private long getDefaultCacheTimeout() {
+        return Long.valueOf(ServerConfiguration.getInstance().getFirstProperty(APIConstants.DEFAULT_CACHE_TIMEOUT))
+                * 60;
+    }
+
+    /**
+     * Returns whether the gateway token cache is enabled.
+     *
+     * @return true if the gateway token cache is enabled
+     */
+    private boolean isGatewayTokenCacheEnabled() {
+        try {
+            APIManagerConfiguration config = getApiManagerConfiguration();
+            String cacheEnabled = config.getFirstProperty(APIConstants.GATEWAY_TOKEN_CACHE_ENABLED);
+            return Boolean.parseBoolean(cacheEnabled);
+        } catch (Exception e) {
+            log.error("Did not found valid API Validation Information cache configuration." +
```

Comment:
Execption being swallowed again leading to original stack trace going missing. Should be,

`log.error("error string", e);`

Reply:
Updated